cli/index.ts

```diff
@@ -445,9 +449,20 @@ function getSnowflakeCredentials(): dataform.ISnowflake {
   };
 }
 
+function graphHasErrors(graph: dataform.ICompiledGraph) {
+  return (
+    graph.graphErrors &&
+    graph.graphErrors.compilationErrors &&
+    graph.graphErrors.compilationErrors.length > 0 &&
```

Comment:
should be `||` (i.e. might have validation errors but no compilation errors, and vice versa)